### PR TITLE
Add Workflow for Deploying to Staging

### PR DESCRIPTION
**Pull request by mzappitello**
* add an action that uses MBTA actions for building and pushing to ECR and deploying to ECS.
* use that action in a workflow that deploys to staging on completed CI on main or via manual trigger.

### CODE REVIEW

File: .github/actions/deploy/action.yaml
Comment (mzappitello) [inline .github/actions/deploy/action.yaml:32]: i looked into this action a bit.

https://github.com/mbta/actions/blob/bb8457c70e8feccf57deb4f0d4a970beef4f2647/deploy-ecs/deploy.sh#L102 checks the status of the current service to decide if an existing process has been spun down and the new one has been spun up.

when running
`aws ecs describe-services --cluster dmap-import --services dmap-import-staging`
locally, i see a the portions of json (picked up [here](https://github.com/mbta/actions/blob/bb8457c70e8feccf57deb4f0d4a970beef4f2647/deploy-ecs/deploy.sh#L27-L30)) all being set to zero. this should cause this [while loop](https://github.com/mbta/actions/blob/main/deploy-ecs/deploy.sh#L100) to exit.

the setup and teardown portion of the action should work as expected with this configuration.

File: .github/workflows/deploy_to_staging.yaml
Comment (mzappitello) [inline .github/workflows/deploy_to_staging.yaml:21]: these secrets need to be added. i don't have the privileges to do that.

File: .github/actions/deploy/action.yaml
Comment (paulswartz) [reply .github/actions/deploy/action.yaml:32]: Yes, but https://github.com/mbta/actions/blob/bb8457c70e8feccf57deb4f0d4a970beef4f2647/deploy-ecs/deploy.sh#L28 will wait for at least 1 task to be running.

File: .github/workflows/deploy_to_staging.yaml
Comment (paulswartz) [reply .github/workflows/deploy_to_staging.yaml:21]: I've given you access to do that.

File: .github/workflows/deploy_to_staging.yaml
Comment (paulswartz) [inline .github/workflows/deploy_to_staging.yaml:4]: praise: neat!

File: .github/actions/deploy/action.yaml
Comment (paulswartz) [inline .github/actions/deploy/action.yaml:2]: nitpick: this does more than build the image and push it to ECR.

Comment (mzappitello) [reply .github/actions/deploy/action.yaml]: updated. i started this before diving into the deploy work.

**Review by paulswartz — COMMENTED** [review]
Other than the actual deploy steps, I think this looks fine. However, as I mentioned in [this comment](https://github.com/mbta/dmap-import/pull/5#discussion_r1384182801) I don't think the actual deploy will work with a task.

**Comment by mzappitello** [comment]
ah. i missed that on my first pass.

i'm assuming there was a reason in another app that caused you to make this [change](https://github.com/mbta/actions/commit/fc343114bbd8b5ff3ab5c2657b6d1267028d7039), so taking that out probably doesn't make a lot of sense.

i'm going to open up a PR that allows us to skip the step entirely in the bash script and run my action against that.

File: .github/actions/deploy/deploy.sh
Comment (mzappitello) [inline .github/actions/deploy/deploy.sh:1]: wrote a smaller version of the deploy script, largely just copying what we already have that we need for this deploy.

File: .github/actions/deploy/deploy.sh
Comment (mzappitello) [inline .github/actions/deploy/deploy.sh:17]: hard coded here.

File: .github/actions/deploy/deploy.sh
Comment (mzappitello) [inline .github/actions/deploy/deploy.sh:33]: we'll have secrets in this project, so i removed the env var used to toggle this check on and off.

File: .github/actions/deploy/deploy.sh
Comment (mzappitello) [inline .github/actions/deploy/deploy.sh:47]: always using fargate, so i removed the else if sections and kept only the `register-task-definition` portion from inside the if fargate block.

File: .github/actions/deploy/deploy.sh
Comment (paulswartz) [inline .github/actions/deploy/deploy.sh:54]: suggestion: I think you also need to update the scheduled task. Looking at it in the [ECS Console](https://us-east-1.console.aws.amazon.com/ecs/v2/clusters/dmap-import/scheduled-tasks/run-dmap-import-staging?region=us-east-1&selectedTarget=ecs-dmap-import-staging) I see that it's configured to use the template task.

**Review by paulswartz — APPROVED** [review]
I don't think this is going to work (see my comment inline), but once it's merged you'll be able to iterate on it from a branch.

**Review by rymarczy — APPROVED** [review]
I'll defer to @paulswartz on this.

Is it possible to also be able to manually run the instance with event bridge? I could see that functionality being useful.

File: .github/actions/deploy/deploy.sh
Comment (mzappitello) [inline .github/actions/deploy/deploy.sh:54]: removed checks that the running task count is up to 1 and check that older deployments have been been stopped. that shouldn't be a concern for us as the task count is always going to be 0 around deployments.

i did think that we could check that the deployment is complete by looking at the `ecs describe-service` command, which results in something like

```
❯ aws ecs describe-services --cluster lamp --services lamp-ingestion-staging | jq -r '.services'
[
  {
    "serviceArn": "arn:aws:ecs:us-east-1:434035161053:service/lamp/lamp-ingestion-staging",
    "serviceName": "lamp-ingestion-staging",
    "clusterArn": "arn:aws:ecs:us-east-1:434035161053:cluster/lamp",
    "loadBalancers": [],
    "serviceRegistries": [],
    "status": "ACTIVE",
    "desiredCount": 1,
    "runningCount": 1,
    "pendingCount": 0,
    "launchType": "FARGATE",
    "platformVersion": "LATEST",
    "platformFamily": "Linux",
    "taskDefinition": "arn:aws:ecs:us-east-1:434035161053:task-definition/lamp-ingestion-staging:91",
    "deploymentConfiguration": {
      "deploymentCircuitBreaker": {
        "enable": false,
        "rollback": false
      },
      "maximumPercent": 100,
      "minimumHealthyPercent": 0
    },
    "deployments": [
      {
        "id": "ecs-svc/8666230551424498921",
        "status": "PRIMARY",
        "taskDefinition": "arn:aws:ecs:us-east-1:434035161053:task-definition/lamp-ingestion-staging:91",
        "desiredCount": 1,
        "pendingCount": 0,
        "runningCount": 1,
        "failedTasks": 0,
        "createdAt": "2023-11-01T15:39:42.776000-04:00",
        "updatedAt": "2023-11-01T15:43:52.868000-04:00",
        "launchType": "FARGATE",
        "platformVersion": "1.4.0",
        "platformFamily": "Linux",
        "networkConfiguration": {
          "awsvpcConfiguration": {
            "subnets": [
              "subnet-0ecf10ef54256e988",
              "subnet-09525d49bf33e8caa",
              "subnet-0b00903afc1dc8019"
            ],
            "securityGroups": [
              "sg-0206b82c4667d6560"
            ],
            "assignPublicIp": "DISABLED"
          }
        },
        "rolloutState": "COMPLETED",
        "rolloutStateReason": "ECS deployment ecs-svc/8666230551424498921 completed."
      }
    ],
```
and we can compare the desk definition with the task definition of the only deployment and that the deployment is at a "COMPLETED" rollout.

**Pull request merged**
